File: Pilhas/Pilha Estatica/pilhaEstatica.java
public class pilhaEstatica implements Empilhavel {
    private int ponteiroTopo;
    private Object[] dados;
    private int tamanho;

    public pilhaEstatica() {
        this(10);
    }

    public pilhaEstatica(int tamanho) {
        ponteiroTopo = -1;
        this.tamanho = tamanho;
        dados = new Object[tamanho];
    }

    public void empilhar(Object elemento) {
        if (!estaCheia()) {
            ponteiroTopo++;
            dados[ponteiroTopo] = elemento;
        } else {
            System.out.println("Stack is full!");
        }
    }

        public Object desempilhar() {
            Object elementoTopo = null;
            if(!estaVazia()) {
                elementoTopo = dados[ponteiroTopo];
                ponteiroTopo--;
            } else {
                System.out.println("Stack is empty!");
            }
            return elementoTopo;
        }


        public Object topo() {
            Object elementoTopo = null;
            if(!estaVazia()) {
                elementoTopo = dados[ponteiroTopo];
                ponteiroTopo--;
            } else {
                System.out.println("Stack is empty!");
            }
            return elementoTopo;
        }


        public boolean estaCheia() {
            return (ponteiroTopo == tamanho - 1);
        }
        public boolean estaVazia() {
            return (ponteiroTopo == -1);
        }
        public String imprimir() {
            String resultado = "[";
            for (int i = 0; i<= ponteiroTopo; i++) {
                resultado += dados[i]+ " ";
            }
            return resultado + "]";
        }
    }

File: Pilhas/Pilha Estatica/PilhaEstatica.kt
class PilhaEstatica : Empilhavel {
    private var dados: Array<Any?>
    private var ponteiroTopo: Int

    constructor(tamanho: Int) {
        dados = arrayOfNulls(tamanho)
        ponteiroTopo = -1
    }

    override fun empilhar(dado: Any?) {
        if (!estaCheia()) {
            ponteiroTopo++
            dados[ponteiroTopo] = dado
        } else {
            println("ERROR! Pilha está cheia!")
        }
    }

    override fun topo(): Any? {
        var dadoTopo: Any? = null
        if (!estaVazia()) {
            dadoTopo = dados[ponteiroTopo]
        } else {
            println("Pilha está vazia!")
        }
        return dadoTopo
    }

    override fun desempilhar(): Any? {
        var dadoTopo: Any? = null
        if (!estaVazia()) {
            dadoTopo = dados[ponteiroTopo]
            ponteiroTopo--
        } else {
            println("Pilha está vazia!")
        }
        return dadoTopo
    }

    override fun estaCheia(): Boolean {
        return ponteiroTopo == dados.size - 1
    }

    override fun estaVazia(): Boolean {
        return ponteiroTopo == -1
    }

    override fun imprimir(): String {
        var resultado = "["
        for (i in ponteiroTopo downTo 0) {
            resultado += "${dados[i]}"
        }
        return "$resultado]"
    }
}
